### PR TITLE
[Dist] Add the GstMQTT element to nnstreamer-misc

### PR DESCRIPTION
**Pull request by wooksong**
This PR adds the GstMQTT element to the nnstreamer-misc package for Tizen and Ubunu/Debian.

Signed-off-by: Wook Song <wook16.song@samsung.com>

**Self evaluation:**
1. Build test: [*]Passed [ ]Failed [ ]Skipped
2. Run test: [*]Passed [ ]Failed [ ]Skipped

### CODE REVIEW

File: packaging/nnstreamer.spec [hunk 1]
```diff
@@ -31,7 +31,7 @@
 %define		grpc_support 1
 %define		pytorch_support 0
 %define		caffe2_support 0
-%define		mqtt_support 0
+%define		mqtt_support 1
```
Comment (jaeyun-jung) [inline packaging/nnstreamer.spec:34]: plz consider VD / DA support (line 85 and 103)

Comment (wooksong) [reply packaging/nnstreamer.spec:34]: Thanks. It is disabled for the DA and TV profiles.

File: packaging/nnstreamer.spec [hunk 2]
```diff
@@ -529,6 +531,8 @@ NNStreamer developer utilities include nnstreamer configuration checker.
 
 %package misc
 Summary:	NNStreamer extra packages
+BuildRequires:	pkgconfig(paho-mqtt-c)
```
Comment (dongju-chae) [inline packaging/nnstreamer.spec:534]: How about using `%if 0%{mqtt_support}` like others.

Comment (wooksong) [reply packaging/nnstreamer.spec:534]: Right. Thanks!

**Review by dongju-chae — APPROVED** [review]
LGTM

**Review by kimjh12 — APPROVED** [review]
LGTM..!!

File: packaging/nnstreamer.spec [hunk 3]
```diff
@@ -987,6 +993,7 @@ cp -r result %{buildroot}%{_datadir}/nnstreamer/unittest/
 
 %files misc
 %{gstlibdir}/libgstjoin.so
+%{gstlibdir}/libgstmqtt.so
```
Comment (jaeyun-jung) [inline packaging/nnstreamer.spec:996]: this also requires condition `%if 0%{?mqtt_support}`

**Comment by myungjoo** [comment]
```
[  323s] [----------] 3 tests from testMqttSrc
[  323s] [ RUN      ] testMqttSrc.src_pull_wrongurl_n
[  323s] [       OK ] testMqttSrc.src_pull_wrongurl_n (0 ms)
[  323s] [ RUN      ] testMqttSrc.src_get_set_properties
[  323s] 
[  323s] (unittest_mqtt:1878090): GLib-GObject-WARNING **: 07:01:26.491: g_object_set_is_valid_property: object class 'GstMqttSrc' has no property named 'testport'
[  323s] [       OK ] testMqttSrc.src_get_set_properties (0 ms)
[  323s] [ RUN      ] testMqttSrc.src_get_set_properties_n
[  323s] 
[  323s] (unittest_mqtt:1878090): GLib-GObject-WARNING **: 07:01:26.492: value "0" of type 'gint64' is invalid or out of range for property 'sub-timeout' of type 'gint64'
[  323s] /home/abuild/rpmbuild/BUILD/nnstreamer-1.7.1/packaging/run_unittests_binaries.sh: line 20: 1878090 Segmentation fault      (core dumped) ${entry} --gtest_output="xml:${entry##*/}.xml"
[  323s] ~/rpmbuild/BUILD/nnstreamer-1.7.1
[  323s] error: Bad exit status from /var/tmp/rpm-tmp.K7Kgsq (%build)
```
?

File: packaging/nnstreamer.spec (hunk 3 above)
Comment (wooksong) [reply packaging/nnstreamer.spec:1001]: @jaeyun-jung Fixed. Thanks!

**Review by jaeyun-jung — APPROVED** [review]

**Comment by myungjoo** [comment]
https://developer.gnome.org/gobject/stable/gobject-The-Base-Object-Type.html#g-object-set

When you set 64bit values with g_object_set, you need some treatment. This appears to be the cause of segfault.

**Comment by wooksong** [comment]
> https://developer.gnome.org/gobject/stable/gobject-The-Base-Object-Type.html#g-object-set
> 
> When you set 64bit values with g_object_set, you need some treatment. This appears to be the cause of segfault.

@myungjoo Thanks. It is applied.

**Review by myungjoo — COMMENTED** [review]
Please make sure that mqtt-unit-test is executed ONLY if mqtt is enabled.
(and make sure that mqtt-unit-test is executed if mqtt is enabeld)

**Comment by wooksong** [comment]
> Please make sure that mqtt-unit-test is executed ONLY if mqtt is enabled.
> (and make sure that mqtt-unit-test is executed if mqtt is enabeld)

@myungjoo https://github.com/nnstreamer/nnstreamer/blob/9853457eef5968a9658d638ab911789f974a293e/tests/meson.build#L120 

It would be guarded by ```mqtt_support_is_available```.

**Pull request merged**